File: Python/504. Base 7.py
class Solution:
    def convertToBase7(self, num):
        """
        :type num: int
        :rtype: str
        """
        if num == 0:
            return str(num)
        ans = list()
        tmp = list()
        if num < 0:
            ans.append('-')
            num = -num
        while num:
            tmp.append(str(num%7))
            num = num//7
        ans.extend(tmp[::-1])
        return ''.join(ans)

File: C++/504. Base 7.cpp
class Solution {
public:
    string convertToBase7(int num) {
        if(num==0) return "0";
        string ans;
        string tmp;
        if(num<0)
        {
            num = -num;
            ans += "-";
        }
        while(num)
        {
            tmp += (char)(num%7+'0');
            num /= 7;
        }
        reverse(tmp.begin(),tmp.end());
        return ans+tmp;
    }
};